tsdb/engine/tsm1/reader.go

```diff
@@ -1233,6 +1233,10 @@ func (d *indirectIndex) UnmarshalBinary(b []byte) error {
 			return fmt.Errorf("indirectIndex: not enough data for index entries count")
 		}
 		count := int32(binary.BigEndian.Uint16(b[i : i+indexCountSize]))
+
+		if count <= 0 {
+			return fmt.Errorf("indirectIndex: the count should have >0 value")
```

Comment:
a better message might include the value that was passed in; something like:

`fmt.Errorf("invalid index %d; count should be greater than 0", count)`

Reply:
LGTM